.github/workflows/build.yml

```diff
@@ -119,11 +119,11 @@ jobs:
           coveralls --service=github
 
   build-and-test-mac-gui:
-    runs-on: macos-latest
+    runs-on: macos-13
     env:
       # We need the official Python, because the GA ones only support newer macOS versions
       # The deployment target is picked up by the Python build tools automatically
-      PYTHON_VERSION: 3.11.1
+      PYTHON_VERSION: 3.12.4
```

Comment:
Question: why do we only need to update the Python version for macOS?

Reply:
The workflow errors that were causing problems were related to the macOS builds, which is why there are no updates to the Windows workflow in this PR. There's a bug filed to improve version coverage in the test matrix, but this here is mainly for the builds to work.

The macos-13 GA runner ships with Python 3.12, and there's also an issue with the virtual environment on macOS. This just fixes that issue by matching the Python version, though I also just filed a bug to track the virtual env bug [separately](https://bugzilla.mozilla.org/show_bug.cgi?id=1902350) (which may help answer your question too.)

Reply:
Makes sense, thanks for the details. :)